streamlit_app/app.py

```
@@ -0,0 +1,263 @@
+def _accept_upload(save_path: Path, columns: list[str]) -> None:
+    row_count = _count_csv_rows(save_path)
+    has_gold = _has_gold_label(columns)
+    st.session_state.dataset_path = save_path
+    st.session_state.dataset_has_gold = has_gold
+    st.caption(
+        f"{row_count} rows · columns: {', '.join(columns)}"
+        + (" · gold_label present — evaluation metrics will be shown" if has_gold else "")
+    )
+
+
+def _render_upload_section(is_running: bool) -> None:
```

Ruff (0.15.12)

[warning] Boolean-typed positional argument in function definition

(FBT001)